app/controllers/concerns/flashable.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+module Flashable
+  extend ActiveSupport::Concern
+
+  def message_for_notice(invalid_records)
+    msg = "Uploading #{upload_params[:csv_type]} succeeded"
+    msg + (invalid_records.present? ? ' with warnings:' : '.')
+  end
+
+  def errors_for_alert(invalid_records)
+    error_list = invalid_records[0, 15].map do |record|
+      line = record.errors.delete(:base)
```

Comment:
why do you want to remove any errors? Shouldn't this be done at the model level maybe by not calling the validation that adds whatever the error is in the first place? I can't help to think that maybe this is a model level concern and not a controller one.